custom_components/smartthinq_sensors/sensor.py

```diff
@@ -297,6 +312,78 @@
     DeviceType.WASHER,
 ]
 
+RANGE_SENSORS = {
+    DEFAULT_SENSOR: {
+        ATTR_MEASUREMENT_NAME: "Default",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FN: lambda x: x._power_state,
+        ATTR_ENABLED: True,
+    },
+    FEAT_COOKTOP_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FN: lambda x: x._cooktop_state,
+        ATTR_ENABLED: True,
+    },
+    FEAT_COOKTOP_LEFT_FRONT_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop Left Front",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FEAT: FEAT_COOKTOP_LEFT_FRONT_STATE,
+    },
+    FEAT_COOKTOP_LEFT_REAR_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop Left Rear",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FEAT: FEAT_COOKTOP_LEFT_REAR_STATE,
+    },
+    FEAT_COOKTOP_CENTER_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop Center",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FEAT: FEAT_COOKTOP_CENTER_STATE,
+    },
+    FEAT_COOKTOP_RIGHT_FRONT_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop Right Front",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FEAT: FEAT_COOKTOP_RIGHT_FRONT_STATE,
+    },
+    FEAT_COOKTOP_RIGHT_REAR_STATE: {
+        ATTR_MEASUREMENT_NAME: "Cooktop Right Rear",
+        ATTR_ICON: DEFAULT_ICON,
+        ATTR_VALUE_FEAT: FEAT_COOKTOP_RIGHT_REAR_STATE,
+    },
+    FEAT_OVEN_STATE: {
```

Comment:
This probably is better to define as `binary_sensor`. It is only on/off and HA history is more readable

Reply:
The cooktops are not on/off. They can have multiple states (off, cooking, paused, ...)

Reply:
Just talking about `FEAT_COOKTOP_STATE` and 'FEAT_OVEN_STATE'. If you insert into `RANGE_BINARY_SENSORS` they will be created as binary_sensor, this is better for sensors that only return 2 state (on and off). You can see for example the door status of the refrigerator.

Reply:
Yes, I have moved those to binary sensors.